File: cama.py
import pandas as pd

class Cama:
    def __init__(self, numero_cama):
        self.numero_cama = numero_cama
        # Abrir el archivo con 'open()' y manejarlo correctamente
        with open("Archivos/camas.csv", "r") as camasfile:
            for linea in camasfile:
                lineaSeparada = linea.strip().split(",")
                numeroCama = lineaSeparada[0]
                if numeroCama == self.numero_cama:
                    self.tipoCama = lineaSeparada[1]
                    # Asumimos que ocupacion es una cadena que dice 'True' o 'False'
                    self.ocupacion = lineaSeparada[2] == 'True'  # Convertir a booleano
                    self.precio = lineaSeparada[4]

    def consultarDisponibilidadCama(self):
        if not self.ocupacion:
            print("La cama se encuentra disponible y se le va a asignar al paciente")
            return True
        else: 
            print("La cama NO se encuentra disponible, asigne otra cama")
            return False

    @staticmethod
    def consultarDisponibilidadGeneral():
        with open("Archivos/camas.csv", "r") as camasfile:
            for linea in camasfile:
                lineaSeparada = linea.strip().split(",")
                numeroCama = lineaSeparada[0]
                tipoCama = lineaSeparada[1]
                ocupacion = lineaSeparada[2] == 'True'  # Convertir a booleano
                precio = lineaSeparada[3]

                if not ocupacion:
                    print(f"La cama {numeroCama} que es {tipoCama} se encuentra desocupada y tiene el precio ${precio}")

    @staticmethod
    def BuscarCamaPorTipo(tipoCama):
        ruta_camas = "Archivos/camas.csv"
        
        # Leer el archivo camas.csv en un DataFrame
        df = pd.read_csv(ruta_camas)
        
        # Filtrar para encontrar la primera cama que cumpla con el tipo y esté libre (ocupacion == False)
        cama_disponible = df[(df['TipoCama'] == tipoCama) & (df['Ocupada'] == False)]
        
        if not cama_disponible.empty:
            # Obtener el índice de la primera cama disponible y actualizar ocupación a True
            index_cama = cama_disponible.index[0]
            df.at[index_cama, 'Ocupacion'] = True
            numero_cama = df.at[index_cama, 'NumeroCama']
            
            # Guardar el DataFrame de nuevo en el archivo CSV
            df.to_csv(ruta_camas, index=False)
            
            print(f"El paciente fue asignado a la cama N°{numero_cama}.")
            return numero_cama
        else:
            print("No se encontraron camas disponibles de ese tipo.")
            return None
        
    def liberar_cama(self):
        if self.ocupacion:
            self.ocupacion = False  # Corregir la asignación, no comparación
            print(f"La cama {self.numero_cama} ha sido liberada.")

File: Csharp/HealthTech/Cama.cs
namespace HealtTech
{
    using System;
    using System.Collections.Generic;
    using System.IO;
    using System.Linq;

    public class Cama
    {
        public string NumeroCama { get; set; }
        public string TipoCama { get; set; }
        public bool Ocupacion { get; set; }
        public string Precio { get; set; }

        private string rutaCamas = @"C:\Users\Luis Angel Seoanes O\OneDrive\Escritorio\HealtTech\Csharp\Archivos\camas.csv";

        // Constructor para inicializar una cama específica
        public Cama(string numeroCama)
        {
            NumeroCama = numeroCama;

            // Leer el archivo de camas
            using (var reader = new StreamReader(rutaCamas))
            {
                while (!reader.EndOfStream)
                {
                    var linea = reader.ReadLine().Split(',');
                    var numero = linea[0];
                    if (numero == NumeroCama)
                    {
                        TipoCama = linea[1];
                        Ocupacion = linea[2].ToLower() == "true"; // Convertir a booleano
                        Precio = linea[3];
                    }
                }
            }
        }

        // Método para consultar disponibilidad de una cama específica
        public bool ConsultarDisponibilidadCama()
        {
            if (!Ocupacion)
            {
                Console.WriteLine("La cama se encuentra disponible y se le va a asignar al paciente.");
                return true;
            }
            else
            {
                Console.WriteLine("La cama NO se encuentra disponible, asigne otra cama.");
                return false;
            }
        }

        // Método estático para consultar disponibilidad de todas las camas
        public static void ConsultarDisponibilidadGeneral()
        {
            var rutaCamas = @"C:\Users\Luis Angel Seoanes O\OneDrive\Escritorio\HealtTech\Csharp\Archivos\camas.csv";
            using (var reader = new StreamReader(rutaCamas))
            {
                while (!reader.EndOfStream)
                {
                    var linea = reader.ReadLine().Split(',');
                    var numeroCama = linea[0];
                    var tipoCama = linea[1];
                    var ocupacion = linea[2].ToLower() == "true";
                    var precio = linea[3];

                    if (!ocupacion)
                    {
                        Console.WriteLine($"La cama {numeroCama}, que es {tipoCama}, se encuentra desocupada y tiene un precio de ${precio}.");
                    }
                }
            }
        }

        // Método estático para buscar una cama por tipo
        public static string BuscarCamaPorTipo(string tipoCama)
        {
            var rutaCamas = @"C:\Users\Luis Angel Seoanes O\OneDrive\Escritorio\HealtTech\Csharp\Archivos\camas.csv";
            var lineas = File.ReadAllLines(rutaCamas).ToList();
            for (int i = 1; i < lineas.Count; i++) // Comienza en 1 para saltar el encabezado
            {
                var linea = lineas[i].Split(',');
                if (linea[1] == tipoCama && linea[2].ToLower() == "false")
                {
                    // Marcar la cama como ocupada
                    linea[2] = "true";
                    lineas[i] = string.Join(",", linea);
                    File.WriteAllLines(rutaCamas, lineas);

                    Console.WriteLine($"El paciente fue asignado a la cama N°{linea[0]}.");
                    return linea[0]; // Retornar el número de cama
                }
            }

            Console.WriteLine("No se encontraron camas disponibles de ese tipo.");
            return null;
        }

        // Método para liberar una cama ocupada
        public void LiberarCama()
        {
            if (Ocupacion)
            {
                Ocupacion = false;
                var lineas = File.ReadAllLines(rutaCamas).ToList();
                for (int i = 1; i < lineas.Count; i++) // Comienza en 1 para saltar el encabezado
                {
                    var linea = lineas[i].Split(',');
                    if (linea[0] == NumeroCama)
                    {
                        linea[2] = "false";
                        lineas[i] = string.Join(",", linea);
                        File.WriteAllLines(rutaCamas, lineas);
                        Console.WriteLine($"La cama {NumeroCama} ha sido liberada.");
                        return;
                    }
                }
            }
            else
            {
                Console.WriteLine("La cama ya está liberada.");
            }
        }
    }

}